packages/dicomImageLoader/src/shared/decodeImageFrame.ts

```diff
@@ -372,4 +383,47 @@ function _validateScalingParameters(scalingParameters) {
   }
 }
 
+function createDest(imageFrame, targetBuffer, TypedArrayConstructor) {
+  const { samplesPerPixel } = imageFrame;
+  const { rows, columns } = targetBuffer;
+  const typedLength = rows * columns * samplesPerPixel;
+  const pixelData = new TypedArrayConstructor(typedLength);
+  const bytesPerPixel = pixelData.byteLength / typedLength;
+  return {
+    pixelData,
+    rows,
+    columns,
+    frameInfo: {
+      ...imageFrame.frameInfo,
+      rows,
+      columns,
+    },
+    imageInfo: {
+      ...imageFrame.imageInfo,
+      rows,
+      columns,
+      bytesPerPixel,
+    },
+  };
+}
+
+/** Scales the image frame, updating the frame in place with a new scaled
+ * version of it (in place modification)
+ */
+function scaleImageFrame(imageFrame, targetBuffer, TypedArrayConstructor) {
+  const dest = createDest(imageFrame, targetBuffer, TypedArrayConstructor);
+  if (!dest) {
```

Comment:
I do'nt see any way in the createDes (please rename to createDestination) that it becomes undefined so this if(!dest) will never happen